.github/workflows/ci.yml

```
@@ -57,12 +31,75 @@ jobs:
           --health-timeout 5s
           --health-retries 10
     steps:
-    - uses: actions/checkout@v3.5.0
-    - name: SBT
-      uses: ./.github/actions/sbt
-      env:
-        USE_DOCKER_FOR_TESTS: false
-        ES6_TEST_URL: http://elasticsearch:9200
-        LOCALSTACK_ENDPOINT: http://localstack:4566
-      with:
-        args: clean compile test:compile test
+      - uses: actions/checkout@v3
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+      - name: Kahuna (client-side)
+        working-directory: ./kahuna
+        run: |
+          npm install-clean
+          npm run undist
+          npm test
+          npm run dist
+      - name: SBT
+        uses: ./.github/actions/sbt
+        env:
+          USE_DOCKER_FOR_TESTS: false
+          ES6_TEST_URL: http://elasticsearch:9200
+          LOCALSTACK_ENDPOINT: http://localstack:4566
+        with:
+          args: clean compile test:compile test debian:packageBin
```

Would it help to use [semicolons](https://www.scala-sbt.org/1.x/docs/Howto-Running-Commands.html#Provide+multiple+commands+to+run+consecutively) here to ensure the commands execute sequentially? And/or to use [batch mode](https://www.scala-sbt.org/1.x/docs/offline/Running.html#Batch+mode) to ensure all the commands are treated as a single command? As done [here](https://github.com/guardian/editorial-preferences/blob/main/.github/workflows/ci.yml#L35). Not sure if this is necessary/beneficial when using the `sbt` action.

---

Thanks for those links @samanthagottlieb - TIL what they actually mean, as was only vaguely aware before.

With the commands above, I believe they're already in the order matching the sbt 'lifecycle' so I'm fairly sure they execute sequentially, and personally at least I think they read better without the semicolons, and since there's no spaces in each command I don't think batch mode is needed either - will keep in mind for other things though - thank you

---

I suspect the `;` syntax might be in use elsewhere because the newer syntax for packageBin has a space:
```
[warn] sbt 0.13 shell syntax is deprecated; use slash syntax instead: Debian / packageBin
```

---

I can't seem to get it forward the args via the sbt action (essentially a docker file to provide `graphicsmagick` etc.) when using quotes and semicolons, so I'm going to leave it as is, since it works and is readable (albeit with the warnings you highlighted @davidfurey ).

---

spaces aren't needed for the new-style sbt commands, this is how you'd run newstyle args on the command line (and here)
 
```suggestion
          args: clean compile test:compile test Debian/packageBin
```

batch mode is the name given to running sbt without entering the sbt shell - as these arguments get passed to sbt command line in the custom dockerfile, it looks like this will get executed in batch mode which will run arguments in order. you'd only need semicolons when running inside the sbt shell, and you only need quotes on the command line if you need to pass arguments to a specific command (eg. testOnly)

---

we probably can remove that custom sbt action? the default github image has sbt; the useful bit of that dockerfile is apt-get installing graphicsmagick and exiftool, which we could also do in an earlier step https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/customizing-github-hosted-runners

---

Thanks all for the clarification on semicolons and batch mode!

---

> we probably can remove that custom sbt action? 

thanks for this suggestion @andrew-nowak - I've gone ahead and consolidated that into the main CI job and its simpler now I feel https://github.com/guardian/grid/pull/4146/commits/7b32ea23478b732b846e393e592c6e528dadabef